src/Internal/Proxy/LegacyProxy.php

```diff
@@ -14,6 +14,18 @@
  * Classes are handled through WCPay\Internal\DependencyManagement\DelegateContainer\LegacyContainer.
  */
 class LegacyProxy {
+	/**
+	 * Checks if a function exists.
+	 *
+	 * Use this for WP, WC, and other generic non-native PHP functions.
+	 *
+	 * @param string $name Name of the function.
+	 * @return bool
+	 */
+	public function function_exists( string $name ): bool {
+		return function_exists( $name );
+	}
```

Comment:
> If I understand correctly, there is a need to mock function_exists (PHP core function) in PHPUnit tests.

Correct, mocking the function would allow us to have better code coverage, including conditions.

> However, putting this method here does not sound right as itself is not a legacy of own code in includes. May better move it to a separate util class, such as `Util` or `PhpCoreUtil`?

Maybe we'd need to update some definitions, but so far, I consider "legacy" everything outside `src`. So the legacy proxy is used even for Woo methods and functions, ex. `wc_get_order()`. With that in mind, do you still think it needs to be moved?

Reply:
> Maybe we'd need to update some definitions, but so far, I consider "legacy" everything outside src. 

That's reasonable with this definition. Just improve the phpdoc a bit to tell that LegacyProxy would cover for PHP core functions if needed too.

However, that also means, you can use `LegacyProxy->call_function('function_exists', 'wcs_order_contains_subscription')` and there is no need to add this separate method.

Reply:
Great point! Both are done in https://github.com/Automattic/woocommerce-payments/pull/7333/commits/ad55bcf1ed108dbc4588096429a92c818593ba30